src/AutoRemoteSyscalls.cc

```diff
@@ -418,7 +418,7 @@ static int recvmsg_socket(ScopedFd& sock) {
   msg.msg_iov = &msgdata;
   msg.msg_iovlen = 1;
 
-  if (0 > recvmsg(sock, &msg, 0)) {
+  if (0 > recvmsg(sock, &msg, MSG_DONTWAIT)) {
```

Comment:
I don't know what causality guarantees the kernel makes here. How do we know that the message has arrived in our socket buffer when we make this call? I mean, we're assuming there's no invisible asynchronous buffering right?

Reply:
> I mean, we're assuming there's no invisible asynchronous buffering right?

Right - I've looked at the kernel code and as far as I can tell, for `socketpair` once the sendmsg systemcall completes, the message is guaranteed to be receivable.I'm not sure to what extent we can call that a guarantee - it somebody did add an asynchronous buffer here, I'm not sure Linus would object. It's your call. That change is independent from the rest of this PR - it just makes debugging tests slightly easier if the tracee socket dies for whatever reason (not that I expect that to ever happen in the regular course), since it turns a hang into a properly logged error.

Reply:
I think we should just drop this. We don't touch this code that often; if you do, hopefully you'll remember to patch this in :-).

Reply:
Fair enough, I'll drop it.